l3kernel/l3candidates.dtx

```diff
@@ -434,6 +434,42 @@
 %   mappings.
 % \end{function}
 %
+% \begin{function}[added = 2021-04-28, noTF]
+%   {\seq_put_item:Nnn, \seq_put_item:cnn, \seq_gput_item:Nnn, \seq_gput_item:cnn}
+%   \begin{syntax}
+%     \cs{seq_put_item:Nnn} \meta{seq~var} \Arg{intexpr} \Arg{item}
+%     \cs{seq_put_item:NnnTF} \meta{seq~var} \Arg{intexpr} \Arg{item} \Arg{true code} \Arg{false code}
+%   \end{syntax}
+%   Removes the item of \meta{sequence} at the position given by
+%   evaluating the \meta{integer expression} and replaces it by
+%   \meta{item}.  Items are indexed from $1$ on the left/top of the
+%   \meta{sequence}, or from $-1$ on the right/bottom.  If the
+%   \meta{integer expression} is zero or is larger (in absolute value)
+%   than the number of items in the sequence, the \meta{sequence} is not
+%   modified.  In these cases, \cs{seq_put_item:Nnn} raises an error
```

Comment:
I was pondering a bit if the more often use case is a possible extension of the sequence, i.e., that in case you get past the existing entries all necessary entries are added (as empty values) but it is propably better to keep this separate (after all it is not clear if the default then should be empty or some value or ... so I guess it is fine as it is now.

Reply:
Fully agreed: producing an error now means that we are free to change and implement more complicated processing later on.